### PR TITLE
added filtering journeys controller action with appropriate service m…

### PR DESCRIPTION
**Pull request by roman-pankiv**
…ethods

dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

**Review by vovazavislyak — APPROVED** [review]

File: Car.Backend/Car.UnitTests/Controllers/JourneyControllerTest.cs [hunk 1]
```diff
@@ -191,5 +191,26 @@ public async Task AddJourney_WhenJourneyIsValid_ReturnsOkObjectResult()
...
+            var journeys = Fixture.Create<IEnumerable<Journey>>();
+            var expectedResult = Mapper.Map<IEnumerable<Journey>, IEnumerable<JourneyModel>>(journeys);
+
+            journeyService.Setup(j => j.GetFilteredJourneys(filterModel));
+
+            // Act
+            var result = await journeyController.GetFiltered(filterModel);
+
+            // Assert
+            using (new AssertionScope())
```
Comment (OleksandrAntoniuk) [inline Car.Backend/Car.UnitTests/Controllers/JourneyControllerTest.cs]: I think we can remove AssertionScope

File: Car.Backend/Car.UnitTests/Services/JourneyServiceTest.cs [hunk 2]
```diff
@@ -416,5 +416,22 @@ public async Task AddAsync_WhenJourneyIsNotValid_ReturnsJourneyObject()
...
+            var expectedJourneys = new List<Journey>();
+
+            journeyRepository.Setup(r => r.Query())
+                .Returns(expectedJourneys.AsQueryable().BuildMock().Object);
+
+            // Act
+            var result = await journeyService.GetFilteredJourneys(filter);
+
+            // Assert
+            result.Should().NotBeNull();
```
Comment (OleksandrAntoniuk) [inline Car.Backend/Car.UnitTests/Services/JourneyServiceTest.cs]: you should check that result should be equal to expected journeys

File: Car.Backend/Car.UnitTests/Controllers/JourneyControllerTest.cs [hunk 3]
```diff
@@ -191,5 +191,26 @@ public async Task AddJourney_WhenJourneyIsValid_ReturnsOkObjectResult()
...
+            {
+                result.Should().BeOfType<OkObjectResult>();
+                (result as OkObjectResult)?.Value.Should().NotBeNull();
```
Comment (OleksandrAntoniuk) [inline Car.Backend/Car.UnitTests/Controllers/JourneyControllerTest.cs]: please verify that the result value is equal to the expectedResult

**Review by OleksandrAntoniuk — CHANGES_REQUESTED** [review]

File: Car.Backend/Car.UnitTests/Controllers/JourneyControllerTest.cs (hunk 1 above)
Comment (roman-pankiv) [reply Car.Backend/Car.UnitTests/Controllers/JourneyControllerTest.cs]: Done

File: Car.Backend/Car.UnitTests/Services/JourneyServiceTest.cs (hunk 2 above)
Comment (roman-pankiv) [reply Car.Backend/Car.UnitTests/Services/JourneyServiceTest.cs]: Done

File: Car.Backend/Car.UnitTests/Controllers/JourneyControllerTest.cs (hunk 3 above)
Comment (roman-pankiv) [reply Car.Backend/Car.UnitTests/Controllers/JourneyControllerTest.cs]: Done

**Review by OleksandrAntoniuk — APPROVED** [review]

**Review by OleksandrAntoniuk — APPROVED** [review]

**Review by bohatyrov — APPROVED** [review]

**Pull request merged**